### ISSUE TITLE
[Http-Fundation] HeaderUtils->combine breaks X-Sendfile

### ISSUE THREAD

**Issue by Alsatian67** [symfony/symfony#28627]
**Symfony versions affected**: >= 4.1 only

**Description**  
This [commit](https://github.com/symfony/symfony/commit/b435e80cae24e85d6e0e923ae932b2f79acdbcb2#diff-90a9ba44670369716af27ea5a3ad2daa) broke the X-SendFile mecanism.  

Due to the `HeaderUtils::combine()` (precedently named `HeaderUtils::combineParts()`) function, [here](https://github.com/symfony/symfony/commit/b435e80cae24e85d6e0e923ae932b2f79acdbcb2#diff-90a9ba44670369716af27ea5a3ad2daaR224) `$pathPrefix` is converted to lower case.

If the real path contains upper case characters, it don't work anymore.

**Possible Solution**  
Cancel the commit changes in BinaryFileResponse.
(compare with `substr(strtolower($path), 0, \strlen($pathPrefix)` is not a solution on unix systems)

**Additional context**  
Example of Nginx configuration which wont work (because "John" starts with an uppercase character) : 

```nginx
 location ~ ^/index\.php(/|$) {
        fastcgi_pass unix:/var/run/php/php7.1-fpm.sock;

        ...

        fastcgi_param  HTTP_X-Sendfile-Type     X-Accel-Redirect;
        fastcgi_param  HTTP_X-Accel-Mapping    /home/John/=/protected_files/;

        internal;
    }
```

**Comment by ro0NL** [comment]
ref #24699 cc @c960657

**Comment by c960657** [comment]
Patch available in #28636.
